### PR TITLE
WIP: Uploading to test cmake building and wrapping

### PR DESCRIPTION
**Pull request by Leengit**
A pull request so as to trigger the github actions.  This may need additional pushes before merging into master.

### CODE REVIEW

**Comment by Leengit** [comment]
Because of the  compiler versioning issue this code still does not compile on GitHub.  However, I want to push these changes to upstream nonetheless so that this pull request does not get too massive to review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,52 +1,65 @@
...
+    set(torch_fetch_additional_libraries "torch_cuda" "c10_cuda")
+  endif()
+  if(NOT DEFINED ENV{USE_CUDNN})
+    # We will not use CUDNN on GitHub
+    set(ENV{USE_CUDNN} "0")
+  endif()
+  set(torch_fetch_install_directory "${torch_fetch_SOURCE_DIR}/../torch_fetch-install")
+  set(torch_fetch_build_directory "${torch_fetch_SOURCE_DIR}/build")
+  message(STATUS "Executing: cd ${torch_fetch_SOURCE_DIR}; ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory}")
+  add_custom_target(
```
Comment (dzenanz) [inline CMakeLists.txt:47]: I think this is executed at build time, not at compile time, so surrounding messages are misleading.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,52 +1,65 @@
...
-#   find_package(Torch REQUIRED CONFIG)
-# endif()
-# ")
-#
-# include_directories(${TORCH_INCLUDE_DIRS})
-# # itk_module_target(${_Torch_lib} NO_INSTALL)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
-set(PyTorch_LIBRARIES PyTorch "${TORCH_LIBRARIES}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--no-as-needed")
+list(APPEND PyTorch_LIBRARIES "${TORCH_LIBRARIES}" "${torch_fetch_additional_libraries}")
```
Comment (dzenanz) [inline CMakeLists.txt]: Where does `TORCH_LIBRARIES` come from? Maybe `TORCH_INCLUDE_DIRS` is also provided, so we can avoid manual list on line 50.

File: include/itkTorchImageToImageFilter.h [hunk 3]
```diff
@@ -23,14 +23,16 @@
...
- * \ brief Filters a image by iterating over its pixels.
+ * \brief Filters a image by iterating over its pixels.
  *
  * Filters a image by iterating over its pixels in a multi-threaded way
  * and {to be completed by the developer}.
  *
- * \ ingroup Torch
+ * \sa ImageToImageFilter
+ *
+ * \ingroup PyTorch
```
Comment (dzenanz) [inline include/itkTorchImageToImageFilter.h:35]: Is rename Torch -> PyTorch intentional?

File: include/itkTorchTensorAccessorHelper.h [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+public:
+  static torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel > &
+  FindPixel( torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel > &accessor, const IndexType & /* index */ )
+    {
+    return accessor;
+    }
+};
+
+/** Specializations for VNumberOfSteps > 0 are not strictly necessary
+ * but are implemented for speed. */
```
Comment (dzenanz) [inline include/itkTorchTensorAccessorHelper.h]: Is this an assumption? Or did you try it both ways? If there is no speed difference, less code is preferred.

**Review by dzenanz — COMMENTED** [review]
See in-line feedback.

File: include/itkTorchImageToImageFilter.h (hunk 3 above)
Comment (Leengit) [reply include/itkTorchImageToImageFilter.h:35]: Yes, to go with the module name ITKPyTorch.  (But if that's wrong, please correct me!)

File: include/itkTorchTensorAccessorHelper.h (hunk 4 above)
Comment (Leengit) [reply include/itkTorchTensorAccessorHelper.h]: There are no image/region iterators for TorchImage (yet) so the speed of these accessors could be quite important.  Pending an opportunity to do some speed tests, I'll strip out the extra code.

File: CMakeLists.txt (hunk 2 above)
Comment (Leengit) [reply CMakeLists.txt]: TORCH_LIBRARIES doesn't exists.  Removing it.

**Review by dzenanz — COMMENTED** [review]
Looks good after a glance.

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,52 +1,73 @@
...
+endif()
+
+# Use torch's setup.py to build and locally install it.
+if(NOT DEFINED torch_fetch_install_directory)
+  set(torch_fetch_install_directory "${torch_fetch_SOURCE_DIR}/../torch_fetch-install")
+  set(torch_fetch_build_directory "${torch_fetch_SOURCE_DIR}/build")
+  message(STATUS "Executing: cd ${torch_fetch_SOURCE_DIR}; ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory}")
+  add_custom_target(
+    temp ALL
+    ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory}
```
Comment (thewtex) [inline CMakeLists.txt:49]: Is this necessary? `add_subdirectory` is used below.

File: CMakeLists.txt (hunk 1 above)
Comment (Leengit) [reply CMakeLists.txt:47]: I talked with @thewtex about how to revamp the `CMakeLists.txt` file.  I will make those changes in a different pull request.

File: CMakeLists.txt (hunk 5 above)
Comment (Leengit) [reply CMakeLists.txt:49]: Thank you for the discussion today.  I will make the changes in another pull request.

**Pull request merged**